kustomize/stage/pod/chaos/pod-init-container-failed.yaml

```diff
@@ -0,0 +1,84 @@
+kind: Stage
+metadata:
+  name: pod-init-container-running-failed
+spec:
+  resourceRef:
+    apiGroup: v1
+    kind: Pod
+  selector:
+    matchExpressions:
+    - key: '.metadata.labels.pod-init-container-failed.stage.kwok.x-k8s.io'
+      operator: 'In'
+      values:
+      - 'true'
+    - key: '.metadata.deletionTimestamp'
+      operator: 'DoesNotExist'
+    - key: '.status.phase'
+      operator: 'In'
+      values:
+      - 'Pending'
+  delay:
+    durationFrom:
+      expressionFrom: '.metadata.annotations.pod-init-container-failed.stage.kwok.x-k8s.io/delay'
+    jitterDurationFrom:
+      expressionFrom: '.metadata.annotations.pod-init-container-failed.stage.kwok.x-k8s.io/jitter-delay'
+  next:
+    delete: false
+    statusTemplate: |
+      {{ $now := Now }}
+      {{ $defaultReason := "initContainerError" }}
+      {{ $defaultMessage := "initContainer reported errors" }}
+      {{ $defaultExitCode := 1 }}
+      {{ $initContainerName := .metadata.annotations.pod-init-container-failed.stage.kwok.x-k8s.io/container-name }}
+      {{ $failureReason := or .metadata.annotations.pod-init-container-failed.stage.kwok.x-k8s.io/reason $defaultReason}}
+      {{ $failureMessage := or .metadata.annotations.pod-init-container-failed.stage.kwok.x-k8s.io/message, $defaultMessage}}
+      {{ $failureExitCode := or .metadata.annotations.pod-init-container-failed.stage.kwok.x-k8s.io/exit-code $defaultExitCode}}
+      conditions:
+      - lastProbeTime: null
+        lastTransitionTime: {{ $now | Quote }}
+        status: "False"
+        reason: ""
+        type: Initialized
+      initContainerStatuses:
+      {{ range $index, $item := .spec.initContainers }}
+      {{ $origin := index $root.status.initContainerStatuses $index }}
+      {{ if eq $item.name $initContainerName }}
+      - image: {{ $item.image | Quote }}
+        name: {{ $item.name | Quote }}
+        ready: false
+        restartCount: {{ $origin.restartCount }} 
+        started: false
+        state:
+          terminated:
+            exitCode: {{ $failureExitCode }}
+            finishedAt: {{ $now | Quote }}
+            reason: {{ $failureReason }}
+            message: {{ $failureMessage }}
+            startedAt: {{ $now | Quote }}
+      phase: Failed
+      {{ break }}
+      {{ else }}
+      - image: {{ $item.image | Quote }}
+        name: {{ $item.name | Quote }}
+        restartCount: {{ $origin.restartCount }} 
+        started: false
+        state:
+          terminated:
+            exitCode: 0
+            finishedAt: {{ $now | Quote }}
+            reason: Completed
+            startedAt: {{ $now | Quote }}
+      {{ end }}
+      {{ end }}
+      containerStatuses:
+      {{ range $index, $item := .spec.containers }}
+      - image: {{ $item.image | Quote }}
+        name: {{ $item.name | Quote }}
+        ready: false
+        restartCount: 0
+        started: false
+        state:
+          waiting:
+            reason: PodInitializing
+      {{ end }}
```

Comment:
It didn't work and I made some changes

- There's something wrong with JQ and Go Template syntax.
- This is already running the init container, so conditions and podIP cannot be omitted.

```diff
diff --git a/kustomize/stage/pod/chaos/pod-init-container-failed.yaml b/kustomize/stage/pod/chaos/pod-init-container-failed.yaml
index aaa0a53f..aaef7126 100644
--- a/kustomize/stage/pod/chaos/pod-init-container-failed.yaml
+++ b/kustomize/stage/pod/chaos/pod-init-container-failed.yaml
@@ -8,7 +8,7 @@ spec:
     kind: Pod
   selector:
     matchExpressions:
-    - key: '.metadata.labels.pod-init-container-failed.stage.kwok.x-k8s.io'
+    - key: '.metadata.labels["pod-init-container-running-failed.stage.kwok.x-k8s.io"]'
       operator: 'In'
       values:
       - 'true'
@@ -20,34 +20,43 @@ spec:
       - 'Pending'
   delay:
     durationFrom:
-      expressionFrom: '.metadata.annotations.pod-init-container-failed.stage.kwok.x-k8s.io/delay'
+      expressionFrom: '.metadata.annotations["pod-init-container-running-failed.stage.kwok.x-k8s.io/delay"]'
     jitterDurationFrom:
-      expressionFrom: '.metadata.annotations.pod-init-container-failed.stage.kwok.x-k8s.io/jitter-delay'
+      expressionFrom: '.metadata.annotations["pod-init-container-running-failed.stage.kwok.x-k8s.io/jitter-delay"]'
+  weight: 10000
   next:
-    delete: false
     statusTemplate: |
       {{ $now := Now }}
       {{ $defaultReason := "initContainerError" }}
       {{ $defaultMessage := "initContainer reported errors" }}
       {{ $defaultExitCode := 1 }}
-      {{ $initContainerName := .metadata.annotations.pod-init-container-failed.stage.kwok.x-k8s.io/container-name }}
-      {{ $failureReason := or .metadata.annotations.pod-init-container-failed.stage.kwok.x-k8s.io/reason $defaultReason}}
-      {{ $failureMessage := or .metadata.annotations.pod-init-container-failed.stage.kwok.x-k8s.io/message, $defaultMessage}}
-      {{ $failureExitCode := or .metadata.annotations.pod-init-container-failed.stage.kwok.x-k8s.io/exit-code $defaultExitCode}}
+      {{ $initContainerName := or ( index .metadata.annotations "pod-init-container-running-failed.stage.kwok.x-k8s.io/container-name" ) "" }}
+      {{ $failureReason := or ( index .metadata.annotations "pod-init-container-running-failed.stage.kwok.x-k8s.io/reason" ) $defaultReason }}
+      {{ $failureMessage := or ( index .metadata.annotations "pod-init-container-running-failed.stage.kwok.x-k8s.io/message" ) $defaultMessage }}
+      {{ $failureExitCode := or ( index .metadata.annotations "pod-init-container-running-failed.stage.kwok.x-k8s.io/exit-code" ) $defaultExitCode }}
+      
       conditions:
       - lastProbeTime: null
         lastTransitionTime: {{ $now | Quote }}
         status: "False"
         reason: ""
         type: Initialized
+      - lastTransitionTime: {{ $now | Quote }}
+        status: "False"
+        reason: ""
+        type: Ready
+      - lastTransitionTime: {{ $now | Quote }}
+        status: "False"
+        reason: ""
+        type: ContainersReady
+      
       initContainerStatuses:
       {{ range $index, $item := .spec.initContainers }}
-      {{ $origin := index $root.status.initContainerStatuses $index }}
-      {{ if eq $item.name $initContainerName }}
+      {{ if or ( not $initContainerName ) ( eq $item.name $initContainerName ) }}
       - image: {{ $item.image | Quote }}
         name: {{ $item.name | Quote }}
         ready: false
-        restartCount: {{ $origin.restartCount }} 
+        restartCount: 0
         started: false
         state:
           terminated:
@@ -56,13 +65,11 @@ spec:
             reason: {{ $failureReason }}
             message: {{ $failureMessage }}
             startedAt: {{ $now | Quote }}
-      phase: Failed
-      {{ break }}
       {{ else }}
       - image: {{ $item.image | Quote }}
         name: {{ $item.name | Quote }}
-        restartCount: {{ $origin.restartCount }} 
-        started: false
+        ready: true
+        restartCount: 0
         state:
           terminated:
             exitCode: 0
@@ -71,6 +78,7 @@ spec:
             startedAt: {{ $now | Quote }}
       {{ end }}
       {{ end }}
+      
       containerStatuses:
       {{ range $index, $item := .spec.containers }}
       - image: {{ $item.image | Quote }}
@@ -82,3 +90,8 @@ spec:
           waiting:
             reason: PodInitializing
       {{ end }}
+      
+      hostIP: {{ NodeIPWith .spec.nodeName | Quote }}
+      podIP: {{ PodIPWith .spec.nodeName ( or .spec.hostNetwork false ) ( or .metadata.uid "" ) ( or .metadata.name "" ) ( or .metadata.namespace "" ) | Quote }}
+      phase: Failed
+      startTime: {{ $now | Quote }}
```

```suggestion
apiVersion: kwok.x-k8s.io/v1alpha1
kind: Stage
metadata:
  name: pod-init-container-running-failed
spec:
  resourceRef:
    apiGroup: v1
    kind: Pod
  selector:
    matchExpressions:
    - key: '.metadata.labels["pod-init-container-running-failed.stage.kwok.x-k8s.io"]'
      operator: 'In'
      values:
      - 'true'
    - key: '.metadata.deletionTimestamp'
      operator: 'DoesNotExist'
    - key: '.status.phase'
      operator: 'In'
      values:
      - 'Pending'
  delay:
    durationFrom:
      expressionFrom: '.metadata.annotations["pod-init-container-running-failed.stage.kwok.x-k8s.io/delay"]'
    jitterDurationFrom:
      expressionFrom: '.metadata.annotations["pod-init-container-running-failed.stage.kwok.x-k8s.io/jitter-delay"]'
  weight: 10000
  next:
    statusTemplate: |
      {{ $now := Now }}
      {{ $defaultReason := "initContainerError" }}
      {{ $defaultMessage := "initContainer reported errors" }}
      {{ $defaultExitCode := 1 }}
      {{ $initContainerName := or ( index .metadata.annotations "pod-init-container-running-failed.stage.kwok.x-k8s.io/container-name" ) "" }}
      {{ $failureReason := or ( index .metadata.annotations "pod-init-container-running-failed.stage.kwok.x-k8s.io/reason" ) $defaultReason }}
      {{ $failureMessage := or ( index .metadata.annotations "pod-init-container-running-failed.stage.kwok.x-k8s.io/message" ) $defaultMessage }}
      {{ $failureExitCode := or ( index .metadata.annotations "pod-init-container-running-failed.stage.kwok.x-k8s.io/exit-code" ) $defaultExitCode }}
      
      conditions:
      - lastProbeTime: null
        lastTransitionTime: {{ $now | Quote }}
        status: "False"
        reason: ""
        type: Initialized
      - lastTransitionTime: {{ $now | Quote }}
        status: "False"
        reason: ""
        type: Ready
      - lastTransitionTime: {{ $now | Quote }}
        status: "False"
        reason: ""
        type: ContainersReady
      
      initContainerStatuses:
      {{ range $index, $item := .spec.initContainers }}
      {{ if or ( not $initContainerName ) ( eq $item.name $initContainerName ) }}
      - image: {{ $item.image | Quote }}
        name: {{ $item.name | Quote }}
        ready: false
        restartCount: 0
        started: false
        state:
          terminated:
            exitCode: {{ $failureExitCode }}
            finishedAt: {{ $now | Quote }}
            reason: {{ $failureReason }}
            message: {{ $failureMessage }}
            startedAt: {{ $now | Quote }}
      {{ else }}
      - image: {{ $item.image | Quote }}
        name: {{ $item.name | Quote }}
        ready: true
        restartCount: 0
        state:
          terminated:
            exitCode: 0
            finishedAt: {{ $now | Quote }}
            reason: Completed
            startedAt: {{ $now | Quote }}
      {{ end }}
      {{ end }}
      
      containerStatuses:
      {{ range $index, $item := .spec.containers }}
      - image: {{ $item.image | Quote }}
        name: {{ $item.name | Quote }}
        ready: false
        restartCount: 0
        started: false
        state:
          waiting:
            reason: PodInitializing
      {{ end }}
      
      hostIP: {{ NodeIPWith .spec.nodeName | Quote }}
      podIP: {{ PodIPWith .spec.nodeName ( or .spec.hostNetwork false ) ( or .metadata.uid "" ) ( or .metadata.name "" ) ( or .metadata.namespace "" ) | Quote }}
      phase: Failed
      startTime: {{ $now | Quote }}
```